tiny_dnn/util/parallel_for.h

```diff
@@ -82,6 +86,31 @@ void parallel_for(int begin, int end, const Func &f, int /*grainsize*/) {
   for (int i = begin; i < end; ++i) f(blocked_range(i, i + 1));
 }
 
+#elif defined(CNN_USE_GCD)
+
+template <typename Func>
+void parallel_for(int begin, int end, const Func &f, int grainsize) {
+  int count     = end - begin;
+  int blockSize = grainsize;
+  if (count < blockSize || blockSize == 0) {
+    blockSize = 1;
+  }
+  int blockCount = (count + blockSize - 1) / blockSize;
+  assert(blockCount > 0);
+
+  dispatch_apply(blockCount, dispatch_get_global_queue(QOS_CLASS_DEFAULT, 0),
+                 ^(size_t block) {
```

Comment:
Yes, that's a good point. __OBJC__ won't work though  as it is only defined when actually compiling Objective-C or Objective-C++ and here we are compiling plain C++ (with blocks).

But, there is a non-block variant called dispatch_apply_f that can be used instead. It will look a bit uglier but is better for other platforms.